Formfix/formfixed

swami-hai-ham opened the pull request:
## Summary

Form error validation fixed
Form gets submitted
data on backend ✅
data from backend on /dashboard ✅
go backend model changed to accept ISO date strings
edit form feature works great just some ISO error at projects that I will resolve
UI fixed

krishnaacharyaa commented:
@PatelYash7 @prathamdupare @Sukomal07 kindly review this

PatelYash7 commented:
@swami-hai-ham Can you please Explain What exact things you have done, like i can't understand the commit messages. All have somelocalchanges. Please use proper commit messages for Features you have added or worked on.

krishnaacharyaa commented:
@swami-hai-ham @Sukomal07 iskko merge karllu? 
@swami-hai-ham tera changes hua (push hua kal kaa?)

swami-hai-ham commented:
# summary

added api/skills unauth route
edited datetime type in go model to accept ISO string format
removed process.env.NEXT_PUBLIC_BACKEND_URL || from some 
components because even after giving env variable it was calling backend:8080 instead of localhost:8080
on /form component 
- changed validation mode & added delayError
- logic to submit to backend in expected format
- logic to get data from backend with id from params & to reset form values

removed zod schema of countryCode from user-zod & created new for phone-input & corrected zod schema of date object to iso string

And modified all my form components accordingly with some ui changes 

added phone input component without separate svg folder

added accordion

@krishnaacharyaa yessir kal changes ho gye 

 @PatelYash7 summary

prathamdupare commented:
Can you show a demonstration video and error management?

krishnaacharyaa commented on frontend/app/(auth)/signup/page.tsx:
```diff
@@ -14,7 +14,7 @@ import { signIn } from 'next-auth/react';
 import { Label } from '@/components/ui/label';
 
-const backendUrl = process.env.NEXT_PUBLIC_BACKEND_URL || 'http://localhost:8080';
+const backendUrl = 'http://localhost:8080';
```
Please change it back

Sukomal07 commented:
> @swami-hai-ham @Sukomal07 iskko merge karllu? @swami-hai-ham tera changes hua (push hua kal kaa?)

@swami-hai-ham Bro , Can you show a demonstration video ?

PatelYash7 commented:
@swami-hai-ham phone input is already there in basic details, then why new component for that only ?
